src/editor/lib/EditorControls.js

```diff
@@ -76,12 +83,73 @@ THREE.EditorControls = function (_object, domElement) {
     }
 
     object.position.copy(
-      target.localToWorld(new THREE.Vector3(0, 0, distance * 2))
+      target.localToWorld(
+        new THREE.Vector3(0, center.y + distance * 0.5, distance * 2.5)
+      )
     );
-    object.lookAt(target.getWorldPosition(new THREE.Vector3()));
+    const pos = target.getWorldPosition(new THREE.Vector3());
+    pos.y = center.y;
+
+    object.lookAt(pos);
+
+    // Save end camera position/quaternion
+    scope.transitionCamPosEnd.copy(object.position);
+    scope.transitionCamQuaternionEnd.copy(object.quaternion);
+    // Restore camera position/quaternion and start transition
+    object.position.copy(scope.transitionCamPosStart);
+    object.quaternion.copy(scope.transitionCamQuaternionStart);
+    scope.transitionSpeed = 0.001;
+    scope.transitionProgress = 0;
+    scope.transitioning = true;
+    // The changeEvent is emitted at the end of the transition below
+  };
 
-    scope.dispatchEvent(changeEvent);
+  function easeInOutQuad(t) {
+    return t < 0.5 ? 2 * t * t : -1 + (4 - 2 * t) * t;
+  }
+
+  this.transitioning = false;
+  this.transitionProgress = 0;
+  this.transitionCamPosStart = new THREE.Vector3();
+  this.transitionCamPosEnd = new THREE.Vector3();
+  this.transitionCamQuaternionStart = new THREE.Quaternion();
+  this.transitionCamQuaternionEnd = new THREE.Quaternion();
+  this.transitionSpeed = 0.001;
+  this.fakeComponent = {
+    isPlaying: true,
+    el: { isPlaying: true },
+    tick: (t, delta) => {
+      if (scope.enabled === false) return;
+      if (this.transitioning) {
+        this.transitionProgress += delta * this.transitionSpeed;
+        const easeInOutTransitionProgress = easeInOutQuad(
+          this.transitionProgress
+        );
+
+        // Set camera position
+        object.position.lerpVectors(
+          this.transitionCamPosStart,
+          this.transitionCamPosEnd,
+          easeInOutTransitionProgress
+        );
+
+        object.quaternion.slerpQuaternions(
+          this.transitionCamQuaternionStart,
+          this.transitionCamQuaternionEnd,
+          easeInOutTransitionProgress
+        );
+
+        if (this.transitionProgress >= 1) {
+          this.transitioning = false;
+          object.position.copy(this.transitionCamPosEnd);
+          object.quaternion.copy(this.transitionCamQuaternionEnd);
+          scope.dispatchEvent(changeEvent);
+        }
+      }
+    }
   };
+  // Register the tick function with the render loop
+  AFRAME.scenes[0].addBehavior(this.fakeComponent);
```

Comment:
Note that addBehavior is not a public aframe api. The fakeComponent here with the isPlaying flags works on aframe 1.5.0, but it doesn't seem to work on aframe master, the tick is not called. I know there was some changes to behaviors with component order. I'm investigating it. I'll do another PR once I found something that works on both 1.5.0 and master, if this one is already merged.

Reply:
noted, thanks @vincentfretin for the heads up

Reply:
I tried to register a real component, then you need to add an entity to the scene with this component, then you need to handle for it to play, and keep playing if you switch back and forth between viewer and editor mode. That's more brain wracking than continuing with this hack, at least the changes are isolated there. I fixed it for aframe 1.6.0 in #817